tools/toolchains/arm.py

```diff
@@ -71,7 +77,27 @@ def __init__(self, target, notify=None, macros=None,
             raise NotSupportedException(
                 "this compiler does not support the core %s" % target.core)
 
-        if getattr(target, "default_toolchain", "ARM") == "uARM":
+        if  hasattr(target, "default_lib"):
+            if  target.default_lib == "std":
+                if (
+                    hasattr(target, "supported_c_libs") == False
+                    or "arm" not in target.supported_c_libs
+                    or "std" not in target.supported_c_libs["arm"]
+                ):
+                    raise NotSupportedException(
+                      "std C library option not supported for this target."
+                    )
+            elif target.default_lib == "small":
+                if (
+                    hasattr(target, "supported_c_libs") == False
+                    or "arm" not in target.supported_c_libs
+                    or "small" not in target.supported_c_libs["arm"]
+                ):
+                    raise NotSupportedException(
+                      "small C library option not supported for this target."
+                    )
```

Comment:
This code is repeated twice in arm.py. It is also only slightly modified in gcc.py and arm.py. It is a candidate to go in mbed_tolchains.py as a method.